openwisp_users/admin.py

```diff
@@ -260,6 +260,28 @@ class OwnerInline(BaseOwnerInline):
 class OrganizationAdmin(BaseOrganizationAdmin, BaseAdmin):
     view_on_site = False
     inlines = [OwnerInline]
+    fields = ['id', 'name', 'is_active', 'slug', 'description', 'email', 'url']
+
+    readonly_fields = ['id']
+
+    def _get_fields(self, fields, request, obj=None):
+        """
+        removes readonly_fields in add view
+        """
+        if obj:
+            return fields
+        new_fields = fields[:]
+        for field in self.readonly_fields:
+            if field in new_fields:
+                new_fields.remove(field)
+        return new_fields
+
+    def get_fields(self, request, obj=None):
+        return self._get_fields(self.fields, request, obj)
```

Comment:
until this line, and replace all this block with the following code, which you should read, understand and learn to replicate yourself without my help in the future:

```python
    def id_str(self, obj):
        return str(obj.pk)

    id_str.short_description = 'UUID'

    def get_fields(self, request, obj=None):
        fields = super(OrganizationAdmin, self).get_fields(request, obj)
        if obj:
            fields = fields[:]
            fields.insert(0, 'id_str')
        return fields
```

Reply:
This code doesn't work when trying to change organization is giving error however in organization add works properly but `readonly_field` is visible (it shouldn't) and name of this field is `Id:` not `UUID`

Reply:
```
def id_str(self, obj):
        return str(obj.pk)
```
and I think that `id` in this case isn't primary key because UUID has got letters too

Reply:
do that change and iterate to improve it / fix what's needed, I did that in 5 minutes so it's normal something may be missing, it's on your side to make it work, I'm showing you the right path to implement this feature with as less code as possible

Reply:
:+1: